app/views/admin/organizations/index.html.slim

```diff
@@ -3,19 +3,48 @@
 
 = render 'layouts/admin/tables'
 
-= grid(@organizations_grid, show_filters: :when_filtered) do |g|
+= define_grid(@organizations_grid, show_filters: :when_filtered) do |g|
 
-  - g.blank_slate content_tag(:div, "No records found", class: 'alert alert-info')
+  - g.blank_slate do
+    - @empty_list = true
+    - content_tag(:div, "No records found", class: 'alert alert-info')
 
   - g.column name: 'ID', attribute: 'id'
 
-  - g.column name: 'Name', attribute: 'name'
+  - g.column name: 'Name', attribute: 'name', detach_with_id: :name_filter
 
-  - g.column name: 'City', attribute: 'city'
+  - g.column name: 'City', attribute: 'city', detach_with_id: :city_filter,
+    custom_filter: :auto
 
-  - g.column name: 'Country' do |organization|
+  - g.column name: 'Country', attribute: 'name', assoc: :country,
+    custom_filter: :auto, detach_with_id: :country_filter do |organization|
     - organization.country.try(:name)
 
   - g.column name: 'Created', attribute: 'created_at'
 
   - g.column name: 'Updated', attribute: 'updated_at'
+
+- unless @empty_list
+  header.block
+    div.row.filters
+      h4 Filter by:
+      label.col-lg-1 Name
+      span.col-lg-3
+        = grid_filter(@organizations_grid, :name_filter)
+
+      label.col-lg-1 City
+      div.col-lg-3
+        = grid_filter(@organizations_grid, :city_filter)
+
+      label.col-lg-1 Country
+      div.col-lg-3
+        = grid_filter(@organizations_grid, :country_filter)
```

Comment:
The spacing on these filters doesn't look quite right:

![image](https://cloud.githubusercontent.com/assets/702320/12624922/69b55b16-c4ff-11e5-94b9-51127c3296df.png)

e.g City is closer to the name select than to the city one...

I wonder if it would be easier to put the label and the filter in the same column instead of separate ones.